server/store/middleware/response/response.go

```diff
@@ -37,6 +40,48 @@ func responseAnnounceClient(next tracker.AnnounceHandler) tracker.AnnounceHandle
 			return err.(FailedToRetrievePeers)
 		}
 
+		switch req.Event {
```

Comment:
I think this should be before we get our peers (to reduce the amount of time that the resp.IPv4Peers and resp.IPv6Peers are in memory.